packages/devtools_app/lib/src/screens/debugger/program_explorer_controller.dart

```diff
@@ -105,18 +109,34 @@ class ProgramExplorerController extends DisposableController
     ScriptRef script,
     List<VMServiceObjectNode> nodes,
   ) {
+    final searchCondition = (node) => node.script?.uri == script.uri;
     for (final node in nodes) {
-      final result = node.firstChildWithCondition(
-        (node) => node.script?.uri == script.uri,
-      );
+      final result = node.firstChildWithCondition(searchCondition);
       if (result != null) {
         selectNode(result);
         result.expandAscending();
+        _selectedNodeIndex.value = _calculateNodeIndex(searchCondition);
         return;
       }
     }
   }
 
+  int _calculateNodeIndex(
+      bool matchingNodeCondition(VMServiceObjectNode node)) {
+    var index = 0;
```

Comment:
just to clarify, index is tracking the position of the node in the flat list representation of the tree, is that correct? if so can we add a comment

Reply:
Yep, added a comment